src/distribution/inverse_wishart.rs

```diff
@@ -0,0 +1,439 @@
+    /// use statrs::distribution::InverseWishart;
+    ///
+    /// let w = InverseWishart::new(2.0, DMatrix::from_row_slice(2, 2, &[1.0, 0.0, 0.0, 1.0])).unwrap();
+    /// assert_eq!(w.scale(), &DMatrix::from_row_slice(2, 2, &[1.0, 0.0, 0.0, 1.0]));
+    /// ```
+    pub fn scale(&self) -> &DMatrix<f64> {
+        &self.scale
+    }
+
+    /// Returns the dimensionality of the Inverse Wishart distribution
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use nalgebra::DMatrix;
+    /// use statrs::distribution::InverseWishart;
+    ///
+    /// let w = InverseWishart::new(3.0, DMatrix::from_row_slice(2, 2, &[1.0, 0.0, 0.0, 1.0])).unwrap();
+    /// assert_eq!(w.p(), 2);
+    /// ```
+    pub fn p(&self) -> usize {
+        self.scale.nrows()
+    }
+}
+
+impl MeanN<DMatrix<f64>> for InverseWishart {
+    /// Returns the mean of the Inverse Wishart distribution
+    ///
+    /// # Formula
+    ///
+    /// ```ignore
+    /// ψ/(ν - p - 1)
+    /// ```
+    ///
+    /// where `ν` is the degree of freedom, `ψ` is the scale matrix, and `p` is the dimensionality of the distribution.
+    fn mean(&self) -> Option<DMatrix<f64>> {
+        if self.freedom > self.p() as f64 + 1.0 {
+            Some(&self.scale / (self.freedom - self.p() as f64 - 1.0))
+        } else {
+            None
+        }
+    }
+}
+
+impl VarianceN<DMatrix<f64>> for InverseWishart {
+    /// Returns the variance of the Inverse Wishart distribution
+    /// See [formula](https://en.wikipedia.org/wiki/Inverse-Wishart_distribution#Moments)
+    fn variance(&self) -> Option<DMatrix<f64>> {
+        let p = self.p() as f64;
+
+        Some(self.scale.map_with_location(|i, j, x| {
+            let n1 = ((self.freedom - p + 1.0) * x.powi(2)) + ((self.freedom - p - 1.0) * self.scale[(i, i)] * self.scale[(j, j)]);
+            let n2 = (self.freedom - p) * (self.freedom - p - 1.0) * (self.freedom - p - 1.0) * (self.freedom - p - 3.0);
+            n1 / n2
+        }))
+    }
+}
+
+impl Mode<Option<DMatrix<f64>>> for InverseWishart {
+    /// Returns the median of the Inverse Wishart distribution
+    ///
+    /// # Formula
+    ///
+    /// ```ignore
+    /// ψ/(ν + p + 1)
+    /// ```
+    ///
+    /// where `ν` is the degree of freedom, `ψ` is the scale matrix, and `p` is the dimensionality of the distribution.
+    fn mode(&self) -> Option<DMatrix<f64>> {
+        if self.freedom > self.p() as f64 + 1.0 {
+            Some(&self.scale * (1.0 / (self.freedom + self.p() as f64 + 1.0)))
+        } else {
+            None
+        }
+    }
+}
+
+impl ::rand::distributions::Distribution<DMatrix<f64>> for InverseWishart {
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> DMatrix<f64> {
+        let w = Wishart::new(
+            self.freedom,
+            self.scale.clone().try_inverse().unwrap(), // We already know S is positive definite
+        ).unwrap();
+        let s = w.sample(rng);
+
+        s.cholesky().unwrap().inverse().symmetric_part()
+    }
```

Comment:
Would you consider implementing `sample_iter` to rely on a single conversion to `Wishart` to obtain multiple samples?